### PR TITLE
Accept some failing quotes in some conditions for SWEDISH_CAR

### PR DESCRIPTION
**Pull request by sampettersson**
# Jira Issue: [APP-1801]

## Optional checklist
- [ ] Unit tests written
- [x] Tested locally
- [ ] Codescouted

[APP-1801]: https://hedvig.atlassian.net/browse/APP-1801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/resolvers/quote-cart/quoteCart.ts [hunk 1]
```diff
@@ -50,6 +50,23 @@ import { resolveLocaleFromAcceptLanguage } from '../utils'
...
+const shouldPassFailingQuote = (insuranceType: string, quoteResults: QuoteCreationResult[]): boolean => {
+  /// if insurance is SWEDISH_CAR allow failing quotes to pass if we have atleast one valid quote
+  if (
+    insuranceType === "SWEDISH_CAR" &&
+    quoteResults.filter(quote => quote.res.status === "success").length > 0
+  ) {
+    return true
+  }
+
+  return false
```
Comment (cpiehl1) [inline src/resolvers/quote-cart/quoteCart.ts:67]:
```suggestion
  /// if insurance is SWEDISH_CAR allow failing quotes to pass if we have atleast one valid quote
  return insuranceType === "SWEDISH_CAR" &&
    quoteResults.filter(quote => quote.res.status === "success").length > 0
```

File: src/resolvers/quote-cart/quoteCart.ts [hunk 2]
```diff
@@ -50,6 +50,23 @@ import { resolveLocaleFromAcceptLanguage } from '../utils'
 
 const logger = factory.getLogger('quoteCart')
 
+interface QuoteCreationResult {
+  type: any;
+  res: any;
+}
```
Comment (cpiehl1) [inline src/resolvers/quote-cart/quoteCart.ts:56]: Do we need this interface?

**Review by cpiehl1 — APPROVED** [review]
I think this should work 😅

File: src/resolvers/quote-cart/quoteCart.ts (hunk 2 above)
Comment (simonauner) [reply src/resolvers/quote-cart/quoteCart.ts:56]: If I interpret the existing code `res` is a `QuoteCreationResult` from `underwrites.ts` as well, so the naming should probably also change 🙃

**Comment by simonauner** [comment]
Nice fix!

File: src/resolvers/quote-cart/quoteCart.ts (hunk 2 above)
Comment (sampettersson) [reply src/resolvers/quote-cart/quoteCart.ts]: Removed!

**Pull request merged**